system/system_monitor/src/cpu_monitor/cpu_monitor_base.cpp

```diff
@@ -159,21 +173,44 @@ void CPUMonitorBase::checkUsage(diagnostic_updater::DiagnosticStatusWrapper & st
         for (const pt::ptree::value_type & child3 : statistics.get_child("cpu-load")) {
           const pt::ptree & cpu_load = child3.second;
           bool get_cpu_name = false;
+
+          tier4_external_api_msgs::msg::CpuStatus cpu_status;
+
           if (boost::optional<std::string> v = cpu_load.get_optional<std::string>("cpu")) {
             cpu_name = v.get();
             get_cpu_name = true;
           }
           if (boost::optional<float> v = cpu_load.get_optional<float>("usr")) {
             usr = v.get();
+            if (cpu_name == "all") {
+              cpu_usage.all.usr = usr;
+            } else {
+              cpu_status.usr = usr;
+            }
```

Comment:
I think it is better to assign cpu_status and switch later.
```
cpu_status.usr = usr;
```

Reply:
Modified along this comment.